tmail-backend/blob/secondary-blob-store/src/main/java/com/linagora/tmail/blob/secondaryblobstore/FailedBlobEvents.java

```diff
@@ -0,0 +1,191 @@
+package com.linagora.tmail.blob.secondaryblobstore;
+
+import java.util.List;
+
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.api.BucketName;
+import org.apache.james.core.Username;
+import org.apache.james.events.Event;
+
+public interface FailedBlobEvents {
+    interface BlobEvent extends Event {
+        FailedObjectStorage getFailedObjectStorage();
+    }
+
+    class BlobAddition implements BlobEvent {
+        private BucketName bucketName;
+        private BlobId blobId;
+        private FailedObjectStorage failedObjectStorage;
+
+        public BlobAddition(BucketName bucketName, BlobId blobId, FailedObjectStorage failedObjectStorage) {
+            this.bucketName = bucketName;
+            this.blobId = blobId;
+            this.failedObjectStorage = failedObjectStorage;
+        }
+
+        @Override
+        public Username getUsername() {
+            return Username.of("AAA");
+        }
+
+        @Override
+        public boolean isNoop() {
+            return false;
+        }
+
+        @Override
+        public EventId getEventId() {
+            return Event.EventId.random();
+        }
+
+        public BucketName getBucketName() {
+            return bucketName;
+        }
+
+        public void setBucketName(BucketName bucketName) {
+            this.bucketName = bucketName;
+        }
+
+        public BlobId getBlobId() {
+            return blobId;
+        }
+
+        public void setBlobId(BlobId blobId) {
+            this.blobId = blobId;
+        }
+
+        public FailedObjectStorage getFailedObjectStorage() {
+            return failedObjectStorage;
+        }
+
+        public void setFailedObjectStorage(FailedObjectStorage failedObjectStorage) {
+            this.failedObjectStorage = failedObjectStorage;
+        }
+    }
+
+    class BlobDeletion implements BlobEvent {
```

Comment:
Isn't a `BlobDeletion` event a specific instance of `BlobsDeletion` event with a single item?

By leveraging this we could significantly reduce the boiler plate code going on in here.

Reply:
changed to record to reduce code